.github/scripts/main/main.sh

```diff
@@ -52,6 +52,82 @@ make install
 export PATH="$PREFIX/bin:$PATH"
 opam --version
 
+if [ "$OPAM_DOC" = "1" ]; then
+  rm -rf src_ext/
+  make -C doc html man-html pages
+
+  if [ "$GITHUB_EVENT_NAME" = "pull_request" ]; then
+    . .github/scripts/common/hygiene-preamble.sh
+    diff="git diff $BASE_REF_SHA..$PR_REF_SHA"
+    set +e
+    files="`$diff --name-only --diff-filter=A | grep 'src/.*mli'`"
+    set -e
+    if [ -n "$files" ]; then
+      echo '::group::new module added - checking it'
+      if $diff --name-only --exit-code -- doc/index.html ; then
+        echo '::error new module added but index not updates'
+        echo "$files"
+        exit 3
+      fi
+      echo '::endgroup::new module added - checking it'
+    else
+      echo 'No new module added'
+    fi
+  fi
+
+  files="
+    doc/index.html
+    doc/html/index.html
+    doc/man-html/index.html
+
+    doc/html/opam-admin/index.html
+    doc/html/opam-core/index.html
+    doc/html/opam-devel/index.html
+    doc/html/opam-format/index.html
+    doc/html/opam/index.html
+    doc/html/opam-installer/index.html
+    doc/html/opam-repository/index.html
+    doc/html/opam-solver/index.html
+    doc/html/opam-state/index.html
+
+    doc/pages/About.html
+    doc/pages/About.md
+    doc/pages/Distribution.html
+    doc/pages/Distribution.md
+    doc/pages/External_solvers.html
+    doc/pages/External_solvers.md
+    doc/pages/FAQ.html
+    doc/pages/FAQ.md
+    doc/pages/Install.html
+    doc/pages/Install.md
+    doc/pages/Manual.html
+    doc/pages/Manual.md
+    doc/pages/Packaging.html
+    doc/pages/Packaging.md
+    doc/pages/Specifying_Solver_Preferences.html
+    doc/pages/Specifying_Solver_Preferences.md
+    doc/pages/Tricks.html
+    doc/pages/Tricks.md
+    doc/pages/Upgrade_guide.html
+    doc/pages/Upgrade_guide.md
+    doc/pages/Usage.html
+    doc/pages/Usage.md
+    doc/man-html/opam-init.html"
```

Comment:
Why just this one?

Reply:
First, I've put only few files, Kate in her review proposed all doc/pages files. But now there is only one file for man page. We should have an automatic like what you proposed for html pages.

Reply:
I've updated the PR with the generation of full list of opam manpages files.